### DISCUSSION TITLE
save to browser

### DISCUSSION THREAD

**Discussion by Nikita-Kechaev** [vgalin/html2image#72] [Q&A] [answered]
Greetings! Please help )) I want to use library on server + django @admin.action. So, can I save screenshots to the user on the local machine or in chrome downloads without saving the file on the server? If you understand me of course, I would be glad to have a hint))

**Comment by vgalin** [comment, marked as answer]
Hello, I'm not familiar with Django, but from what I've understand, you would like to send directly the image to the client without it being saved on the server.

Currently, this is not possible, as the screenshots are generated as-is by Chrome/Chromium Headless. I would recommend you to :
1. Screenshot;
2. Send the image to the client;
3. Delete the screenshot on the server.

That's probably the best you can do right now...

In future versions of Html2Image, you'll also have the possibility to take screenshots "differently" (Chrome Headless + Chrome Devtools Protocol), and retrieving only the base-64 encoded image without it being saved on the server could be possible ... But I have no Estimated Time of Arrival to give you for the new version of this package, as "many" (for the time I have on my hands) things still remain to be done.

I hope this answered your question.

**Reply by Nikita-Kechaev** [reply]
Yes, everything is clear! Thank you very much for your advice and have a nice day!
